snapcraft/internal/meta/plugs.py

```diff
@@ -0,0 +1,134 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+
+from collections import OrderedDict
+from snapcraft.internal.meta.errors import PlugValidationError
+from typing import Any, Dict, List, Set  # noqa: F401
+
+logger = logging.getLogger(__name__)
+
+
+class Plug:
+    """Representation of a generic snap plug."""
+
+    def __init__(self, *, plug_name: str = "") -> None:
+        self._plug_dict = dict()  # type: Dict[str, Any]
+        self.plug_name = plug_name
+
+    def validate(self):
+        pass
+
+    @classmethod
+    def from_dict(cls, plug_dict: Dict[str, Any], plug_name: str = ""):
+        # If we explicitly support the type, use it instead.
+        if plug_dict.get("interface") == "content":
```

Comment:
"content" being an enum or better yet a dictionary of interface names holding class handlers may simplify this.

Reply:
I made a dict of interfaces (SLOT_MAPPINGS and PLUG_MAPPINGS), that good?